### PR TITLE
Add Unit tests for UX and Engine packages

### PR DESCRIPTION
**Pull request by kris-gaudel**
Added comprehensive unit tests for UX and Engine packages to improve test coverage and reliability.

Changes:
1. UX Package:
- Added tests for `PrintVersion`, `PrintUsage`, `NewProgressWriter`
- Added tests for all tracker constructors (`NewRuleTracker`, `NewProblemsTracker`, etc.)
- Added tests for error helpers and output functions
- Added tests for utility functions like `ProcessName`, `WriteDataSourceTemplate`
- Coverage increased from 8.0% to 32.0%

2. Engine Package:
- Fixed panic in `StartLinesTracker` by properly initializing UxEval instances
- Added tests for `New`, `AddRules`, `GetCre`, `Close`, and `Run` functions
- Added test cases for edge cases and error conditions
- Coverage increased to 10.9%

Overall test coverage increased from 45.2% to 49.43%

### CODE REVIEW

File: internal/pkg/ux/ux_util_test.go [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+	}
+
+	// Verify file exists
+	if _, err := os.Stat(output); os.IsNotExist(err) {
+		t.Error("Expected file to exist")
+	}
+}
+
+// Helper function to check if a string contains a substring
+func contains(s, substr string) bool {
```
[the remark this answers is not in the document]
Comment (kris-gaudel) [reply internal/pkg/ux/ux_util_test.go]: Sorry, idk how I missed this 😅
